sql/2020/20_Caching/cache_ttl_and_content_age_diff_third_party.sql

```diff
@@ -0,0 +1,39 @@
+#standardSQL
+# Difference between Cache TTL and the content age for third party request
+CREATE TEMPORARY FUNCTION toTimestamp(date_string STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var timestamp = Math.round(new Date(date_string).getTime() / 1000);
+    return isNaN(timestamp) ? -1 : timestamp;
+  } catch (e) {
+    return -1;
+  }
+''';
+
+SELECT
+  client,
+  party,
+  percentile,
+  APPROX_QUANTILES(diff_in_days, 1000)[OFFSET(percentile * 10)] AS diff_in_days
+FROM
+  (
+    SELECT
+      client,
+      IF (STRPOS(NET.HOST(url), REGEXP_EXTRACT(NET.REG_DOMAIN(page), r'([\w-]+)')) > 0, 1, 3) AS party,

```

Comment:
Could this be simplified?
```suggestion
      IF (NET.HOST(url) = NET.HOST(page), 1, 3) AS party,

```

Reply:
Noted, third party queries dropped as not used in Chapter. Will use it in case there is a third-party query.